### PR TITLE
Add user guide documentation for contribution and verification

### PR DESCRIPTION
**Pull request by ping-ke**
---
name: Add user guide
about: Add user guide documentation for contribution and verification
---

# Description

Add user guide documentation:
1. contribution: trusted setup background, requirement, and how to contribute
2. verification: Overview and step-by-step verify

# How Has This Been Tested?
Contribute to the ceremony by following the `contribution.md` documentation, and complete it successfully.
Verify your contribution by following the `verification.md` documentation, and complete it successfully.

### CODE REVIEW

File: docs/contribution.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+participants; thus each produces only a fragment of the information needed to generate the ceremony parameter, 
+the so-called toxic waste. As the name could suggest, this information must be thrown away by the participants, 
+and it only takes at least one of them to successfully forget the toxic waste to ensure the security of the ceremony 
+because an attacker needs all fragments in order to break the system; that’s why these events are called MPC Trusted 
+Setup Ceremonies.
+
+### How does a Trusted Setup work?
+Any Trusted Setup ceremony is organized in two subsequent steps: the Phase 1, is universally reusable in any point of 
+contribution as input for any zkSNARKs Phase 2. Briefly, the goal of the Phase 1 is to securely generate zk-SNARK 
+parameters for circuits of up to a huge amount of constraints while the Phase 2, s a circuit-specific ceremony that
```
Comment (qzhodl) [inline docs/contribution.md]: while the Phase 2, (i)s

File: docs/verification.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+- Verifying the circuit we are securing does not include security holes;
+- Verifying the transcript matches the circuit;
+- Verifying contributors' attestations match the transcript;
+
+## Verifying the Circuit
+Confirming the integrity and security of the [EthStorage blob poseidon circuit](https://github.com/ethstorage/zk-decoder/blob/main/circom/circuits/blob_poseidon_2.circom) by reviewing our circom code, which is open-source and available on Github.
+
+## Verifying the Transcript Matches the Circuit
+Our ceremony transcript is included in the zkey published on ceremony.ethstorage.io in the "Download ZKey" tab of the 
+[EthStorage Circom page](https://ceremony.ethstorage.io/projects/EthStorage%20circom). You can verify it matches the
```
Comment (qzhodl) [inline docs/verification.md:24]: The link is not valid, even for the the http version

File: docs/verification.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+- Generate the `r1cs` file from the `circom` source files, by running `circom blob_poseidon_2.circom --r1cs` in the 
+directory `zk-decoder/circom/circuits`. The SHA-256 hash of this file (i.e. as computed by `sha256sum blob_poseidon_2.r1cs`) is `ba17a94a5edcc1d0b0ad10990e0aea4894f080ce571b0db79fb1c3a9c4a74a0d`
+- Use snarkjs to verify that the transcript matches this circuit and powers of tau, by running 
+`snarkjs zkey verify blob_poseidon_2.r1cs powersOfTau28_hez_final_20.ptau blob_poseidon_2_final.zkey`. You should see 
+a list of contribution hashes (attestations) followed by the message snarkJS: `ZKey Ok!`.
+
+
+## Verifying Contributor Attestations Match the Transcript
+
+Each contributor authorized our ceremony to publish an attestation to their GitHub account. (In a Gist entitled
```
Comment (qzhodl) [inline docs/verification.md:40]: We may need to reproduce the verification process before reviewing the doc

File: docs/verification.md (hunk 2 above)
Comment (ping-ke) [reply docs/verification.md:24]: adding the https.

**Pull request closed without merging**